dsc_lib/src/parser/functions.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use serde_json::Value;
+use tree_sitter::Node;
+
+use crate::DscError;
+use crate::parser::{
+    expressions::Expression,
+    FunctionDispatcher,
+};
+
+#[derive(Clone)]
+pub struct Function {
+    name: String,
+    args: Option<Vec<FunctionArg>>,
+}
+
+#[derive(Clone)]
+pub enum FunctionArg {
+    String(String),
+    Integer(i32),
+    Boolean(bool),
+    Expression(Expression),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum FunctionResult {
+    String(String),
+    Object(Value),
+}
+
+impl Function {
+    /// Create a new `Function` instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `function_dispatcher` - The function dispatcher to use.
+    /// * `statement` - The statement that the function is part of.
+    /// * `function` - The function node.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the function node is not valid.
+    pub fn new(function_dispatcher: &FunctionDispatcher, statement_bytes: &[u8], function: &Node) -> Result<Self, DscError> {
```

Comment:
`function_dispatcher` can be removed now for `new`. It's not really used, but just passed around.